llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -5415,6 +5423,11 @@ void AArch64DAGToDAGISel::Select(SDNode *Node) {
         SelectMultiVectorLuti(Node, 2, Opc, 3);
       return;
     }
+    case Intrinsic::aarch64_sme_luti4_zt_x4: {
+      // Does not have immediate but it has 2ZPR input
+      SelectMultiVectorLuti(Node, 4, AArch64::LUTI4_4ZZT2Z, 0, true);
```

Comment:
No strong opinion, but as `isMultiVector` and `MaxImm` are mutually exclusive parameters (i.e, if you supply one, it makes no sense to supply the other), you could separate this into two separate functions.

Reply:
I create a SelectMultiVectorLutiLane and a SelectMultiVectorLuti.